files/nodebot.rb

```diff
@@ -57,5 +60,101 @@ def handle
   def send(channel, message)
     system('nodebot', channel, message)
   end
+
+end
+
+ANSI_TO_IRC_COLORS = {
+  30 => "01",  # black
+  31 => "04",  # red
+  32 => "09",  # green
+  33 => "08",  # yellow
+  34 => "02",  # blue
+  35 => "13", # pink
+  36 => "11", # cyan
+  37 => "00",  # white
+  39 => "00",   # white
+}
+# Background colors
+ANSI_TO_IRC_BACK = {}
+# Formatting characters
+ANSI_TO_IRC_FMT = {
+  0  => 0xf,  # reset
+  1  => 0x2,  # bold
+  4  => 0x1f, # underline
+  7  => 0x16, # reverse mode
+}
+# Generate background colors
+ANSI_TO_IRC_COLORS.each { |k, v|
```

Comment:
multiline blocks should be wrapped in `do` / `end` and this whole thing belongs closer to where `ANSI_TO_IRC_BACK` is defined